src/app/forms/directives/nesting-level/nesting-level.directive.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Directive, ElementRef, Input, OnInit,
+} from '@angular/core';
+
+@Directive({
+  selector: '[appNestingLevel]',
+})
+export class NestingLevelDirective implements OnInit {
+  @Input() nestingLevel = 0;
+
+  constructor(public readonly elementRef: ElementRef<HTMLElement>) {
+  }
+
+  ngOnInit(): void {
+    const ml = this.nestingLevel > 1 ? (2 * (this.nestingLevel - 1)) : 0;
```

Comment:
what are these consts?

Reply:
lol. Sorry, I've been using tailwind for so long these looked normal: margin left, padding left, margin bottom, padding top, and border width.  Essentially this adds a grey border and indentation proportional to the nesting level, which is hard to do by simply adding a css class, so this modifies css properties directly though js. This directive might be a bit of gold plating, and not actually needed. The designs we were given are subject to change, and they might not use the indentation thing in the final version, we just wanted to ensure that our approach was capable of handling the same level of nesting/complexity as previous versions.